Assets/Scripts/AI/Base/AIManager.cs

```diff
@@ -0,0 +1,184 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+[RequireComponent(typeof(AIAnimationManager))]
+public class AIManager : MonoBehaviour
+{
+    public List<Transform> wayPointList;
+    public GameObject eyes;
+
+    [HideInInspector] public NavMeshAgent navMeshAgent;
+    [HideInInspector] public int nextWayPoint;
+    [HideInInspector] public AIAnimationManager animationManager;
+    [HideInInspector] public Vector3 lastKnownTargetPosition;
+
+
+    private Transform chaseTarget;
+    private MarkerManager markerManager;
+    private bool isInCombat = false;
+    protected bool characterRooted = true;
+    private float attackCooldown;
+    private float timer;
+    private bool isTimerSet;
+    private float health = 100;

```

Comment:
This needs to be editable per enemy type - make a constant 'enemy health' and assign in the start function? 

health = Constants.enemyHealth

Reply:
I think each AI needs his own health, (depending on how strong the Gameplay team want it to be). So it might be better to get the health from a scriptable object. (or public max health)